luigi/contrib/batch.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Outlier Bio, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+AWS Batch wrapper for Luigi
+
+From the AWS website:
+
+    AWS Batch enables you to run batch computing workloads on the AWS Cloud. 
+
+    Batch computing is a common way for developers, scientists, and engineers 
+    to access large amounts of compute resources, and AWS Batch removes the 
+    undifferentiated heavy lifting of configuring and managing the required 
+    infrastructure. AWS Batch is similar to traditional batch computing 
+    software. This service can efficiently provision resources in response to 
+    jobs submitted in order to eliminate capacity constraints, reduce compute 
+    costs, and deliver results quickly.
+
+See `AWS Batch User Guide`_ for more details.
+
+To use AWS Batch, you create a jobDefinition JSON that defines a `docker run`_
+command, and then submit this JSON to the API to queue up the task. Behind the
+scenes, AWS Batch auto-scales a fleet of EC2 Container Service instances, 
+monitors the load on these instances, and schedules the jobs.
+
+This `boto3-powered`_ wrapper allows you to create Luigi Tasks to submit Batch
+``jobDefinition``s. You can either pass a dict (mapping directly to the
+``jobDefinition`` JSON) OR an Amazon Resource Name (arn) for a previously
+registered ``jobDefinition``.
+
+Requires:
+
+- boto3 package
+- Amazon AWS credentials discoverable by boto3 (e.g., by using ``aws configure``
+  from awscli_)
+- An enabled AWS Batch job queue configured to run on a compute environment.
+
+Written and maintained by Jake Feala (@jfeala) for Outlier Bio (@outlierbio)
+
+.. _`docker run`: https://docs.docker.com/reference/commandline/run
+.. _jobDefinition: http://http://docs.aws.amazon.com/batch/latest/userguide/job_definitions.html
+.. _`boto3-powered`: https://boto3.readthedocs.io
+.. _awscli: https://aws.amazon.com/cli
+.. _`AWS Batch User Guide`: http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_GetStarted.html
+
+"""
+
+import json
+import logging
+import random
+import string
+import time
+
+import luigi
+logger = logging.getLogger(__name__)
+
+try:
+    import boto3
+except ImportError:
+    logger.warning('boto3 is not installed. BatchTasks require boto3')
+
+
+class BatchJobException(Exception):
+    pass
+
+
+POLL_TIME = 10
```

Comment:
Thoughts on making this
* configurable in `luigi.cfg` or
* override-able parameter?